jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -221,6 +221,8 @@ def mos_spec1d_parser(app, data_obj, data_labels=None):
     with app.data_collection.delay_link_manager_update():
 
         for i in range(len(data_obj)):
+            # store the mosviz_row (used by viewer_data_select to filter entries)
+            data_obj[i].meta['mosviz_row'] = i
```

Comment:
currently only stores for 1D spectra (but that's all we need it for here since the data menu is hidden for the other viewers)

Reply:
this also is currently causing tests to fail, since the tests pass a list of integers as data objects (and an integer doesn't and can't have a `.meta` attribute).  Should I do a `hasattr` check and just skip this line for the sake of those tests, update the tests to not use integers, or does this logic of tracking the row need to be handled in a different way entirely?

Reply:
Is the code *supposed* to be able to handle list of ints? If not, I think the test is wrong and needs to be fixed.

Reply:
ah, the `add_data` line is what triggers the `TypeError` that the test is expecting... but unfortunately we need to set the row index before `add_data` so that it gets properly passed through the `stack_items` dictionary to vue.  So although its a bit ugly, I think a try/except is the best solution here (but open to other ideas).